File: android/app/src/main/kotlin/run/nuri/getagrip/store/StorageMode.kt
// SPDX-License-Identifier: MPL-2.0
// Original contributions Copyright 2026 Nuri Bruner.

package run.nuri.getagrip.store

import run.nuri.getagrip.engine.L10n

/// Where the routines and session history actually ended up. Shown in Settings › About
/// unconditionally: someone who thinks device-only history is backed up finds out when they
/// lose the device.
///
/// TRANSLATION NOTE (Sources/Store/StorageMode.swift): iOS has three cases (App Group,
/// CloudKit fallbacks). **Android has one**, `localOnly`: no CloudKit, Room in the app's
/// data directory, and Auto Backup (`res/xml/data_extraction_rules.xml`) is a
/// device-to-device restore, not a sync. Kept an enum so a real sync one day is a case, not
/// a refactor.
enum class StorageMode {
    /// The app's own data directory, no sync. Auto Backup may carry it to a new phone;
    /// nothing keeps two phones in step.
    localOnly;

    val aboutLine: String
        get() = when (this) {
            localOnly -> L10n.tr(
                "Your routines and sessions are stored on this device. They don't sync between devices."
            )
        }
}

File: Sources/Store/StorageMode.swift
// SPDX-License-Identifier: MPL-2.0
// Original contributions Copyright 2026 Nuri Bruner.

import Foundation

/// Where the routines and session history actually ended up.
///
/// Surfaced in Settings › About, unconditionally: someone who believes their training
/// history is in iCloud when it is device-only would discover the truth at the moment
/// they lose the device. Asserting the happy case is not a safe default.
enum StorageMode {
    /// App Group container configured for CloudKit private sync. This describes the
    /// store configuration, not account availability or a completed upload.
    case cloud
    /// App Group container, no sync — the iCloud entitlement is missing (ad-hoc
    /// simulator signing strips it) or the CloudKit container could not be opened.
    case localOnly
    /// Neither entitlement available: a DIFFERENT store file at the default location.
    /// Anything previously saved to the group container is not in this one, so an
    /// existing user opens the app to an empty history — it must never pass silently.
    case isolated

    var aboutLine: String {
        switch self {
        case .cloud:
            String(localized: "Stored on this device and synced to your private iCloud when available.")
        case .localOnly:
            String(localized: "iCloud is unavailable. Data stays on this device and won't sync.")
        case .isolated:
            String(localized: "Can't reach shared storage. Earlier routines may be missing.")
        }
    }
}
